pkg/services/secrets/kvstore/plugin.go

```diff
@@ -255,7 +256,7 @@ func SetPluginStartupErrorFatal(ctx context.Context, kvstore *kvstore.Namespaced
 	return kvstore.Set(ctx, QuitOnPluginStartupFailureKey, "true")
 }
 
-func EvaluateRemoteSecretsPlugin(ctx context.Context, mg plugins.SecretsPluginManager, cfg *setting.Cfg) error {
+func EvaluateRemoteSecretsPlugin(ctx context.Context, mg plugins.SecretsPluginManager, cfg *setting.Cfg, hasStarted bool) error {
```

Comment:
This seems like a bit too much burden on people calling this function to know if the plugin should have started or not.

My thought is that we update the SecretsManagerPlugin interface to have a HasStarted func. Maybe whenever `mg.SecretsManager(ctx)` is called, that function should check if `self.SecretsManager != nil` and if so set an internal bool to true. HasStarted() can just return that bool and we can easily track if the plugin has started and then exited.

Reply:
Sounds good! Will do!

Reply:
I moved the check to a separate function on 08c16e4afa but `mg.SecretsManager(ctx)` returns a `*plugins.Plugin`, I'm not sure if we should add the `HasStarted` function to `plugins.Plugin` since many plugins already depend on it, so I added the HasPluginStarted function to the `kvstore` package on `plugin.go` instead, in the end it has the same effect, removing the burden when calling `EvaluateRemoteSecretsPlugin`.